### ISSUE TITLE
Upgrade `marcel` gem to 1.0.3 triggers `ActiveStorage::BlobTest#test_create_and_upload_extracts_content_type_from_filename` failure

### ISSUE THREAD

**Issue by yahonda** [rails/rails#51221]
### Steps to reproduce
Managed to reproduce CI failure https://buildkite.com/rails/rails-nightly/builds/230#018df22a-da45-4d75-a78c-a86fb60fb7b5/1165-1205

```ruby
git clone https://github.com/rails/rails
cd rails
bundle update marcel --conservative
git diff
cd activestorage
bin/test test/models/blob_test.rb -n test_create_and_upload_extracts_content_type_from_filename
```

### Expected behavior
It should pass.

### Actual behavior
It fails as follows.

```ruby
$ git diff
diff --git a/Gemfile.lock b/Gemfile.lock
index f0cc4b9d35..ba1c652f0b 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -317,7 +317,7 @@ GEM
       net-imap
       net-pop
       net-smtp
-    marcel (1.0.2)
+    marcel (1.0.3)
     matrix (0.4.2)
     mdl (0.12.0)
       kramdown (~> 2.3)

$ bin/test test/models/blob_test.rb -n test_create_and_upload_extracts_content_type_from_filename
... snip ...
Run options: -n test_create_and_upload_extracts_content_type_from_filename --seed 33276

# Running:

F

Failure:
ActiveStorage::BlobTest#test_create_and_upload_extracts_content_type_from_filename [test/models/blob_test.rb:51]:
Expected: "text/plain"
  Actual: "application/octet-stream"


bin/test test/models/blob_test.rb:49



Finished in 0.012181s, 82.0920 runs/s, 82.0920 assertions/s.
1 runs, 1 assertions, 1 failures, 0 errors, 0 skips
$
```

### System configuration
**Rails version**: main branch

**Ruby version**: ruby 3.3.0 (2023-12-25 revision 5124f9ac75) [x86_64-linux]

**Comment by yahonda** [comment]
According to git bisect, this behavior has been introduced by https://github.com/rails/marcel/commit/8080070e0cf38fb0e3c5623212924312020a7959 via https://github.com/rails/marcel/pull/94

I do not know if this is an expected behavior change at marcel or not because it changes something about illustrator/pdf, not text

cc @jeremy as the author of https://github.com/rails/marcel/pull/94/commits/574ffecd38b8cf0f30edb7e9b1034725905e89ee

**Comment by jeremy** [comment]
I'll investigate! 🙏🏼

**Comment by jeremy** [comment]
Found the [regression](https://github.com/rails/marcel/pull/94/files#diff-c927ecde16e5ffa50375f5125b22357f6060cbeb7d66109ee809a9e95f1365d0L71).

Fixed in https://github.com/rails/marcel/pull/99 and released in Marcel 1.0.4.
